### PR TITLE
Change fill_events() on the SkyCube

### PR DESCRIPTION
**Pull request by JouvinLea**
@adonath 
As discussed, change to histogram directly on the energy without using the pixtoworld of LogEnergyAxis

### CODE REVIEW

**Comment by JouvinLea** [comment]
@adonath 
can we merge?

**Comment by cdeil** [comment]
I think I'm +1 to treat the energy axis for counts cubes, or more generally cubes that represent integral quantities and have an `EBOUNDS` binning instead of `ENERGY` nodes differently,
either by having a separate cube class, or a separate axis class for that case.

@adonath - Maybe we can discuss concrete options and then make a decision this week?

@JouvinLea - If you keep the diff for pull requests small, this makes them easier to review, and it also means much lower chance of merge conflicts or simpler to resolve merge conflicts, if others touch the file before your PR is merged. So for the future, please, no whitespace changes in unrelated parts of the files. I'll go through and do code formatting every few months before a release, when not many PRs are open.

File: gammapy/cube/core.py [hunk 1]
```diff
@@ -186,10 +190,11 @@ def fill_events(self, events, weights=None):
         """
         if weights is not None:
             weights = events[weights]
+        xx, yy = self.sky_image_ref.wcs_skycoord_to_pixel(events.radec)
+        zz = events.energy
+        bins = self.energy_axis.energy, self.sky_image_ref._bins_pix[0], \
```
Comment (adonath) [inline gammapy/cube/core.py]: Can you replace `self.energy_axis.energy` by a call to `self.energies(mode='edges')`?

File: gammapy/cube/core.py [hunk 2]
```diff
@@ -199,7 +204,8 @@ def _bins_energy(self):
         return np.arange(self.data.shape[0] + 1) - 0.5
```
Comment (adonath) [inline gammapy/cube/core.py]: Can you remove this property `_bins_energy()` its not needed anymore.

**Comment by adonath** [comment]
@cdeil That's a exactly what I'd like to avoid :-) But let's discuss this offline this week...

**Comment by JouvinLea** [comment]
@adonath @cdeil 
Is it possible to merge as it is since it's works now and the histogram binning is done correctly? And then when you will decide how to improve the energy axis of the Cube we can change?

**Comment by adonath** [comment]
@leajouvin Thanks, merged!

**Pull request merged**